magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merging_runs.py

```diff
@@ -0,0 +1,571 @@
+"""
+This script splits the proton MC data sample into "train"
+and "test", deletes possible failed runs (only those files
+that end up with a size < 1 kB), and generates the bash
+scripts to merge the data files calling the script "merge_hdf_files.py"
+in the following order:
+
+MAGIC:
+1) Merge the subruns into runs for M1 and M2 individually.
+2) Merge the runs of M1 and M2 into M1-M2 runs.
+3) Merge all the M1-M2 runs for a given night.
+Workingdir/DL1/Observations/Merged
+
+MC:
+1) Merges all MC runs in a node and save them at
+Workingdir/DL1/MC/PARTICLE/Merged
+
+Usage:
+$ python merging_runs_and_splitting_training_samples.py (-c config.yaml)
+
+If you want to merge only the MAGIC or only the MC data,
+you can do as follows:
+
+Only MAGIC:
+$ python merging_runs_and_splitting_training_samples.py --analysis-type onlyMAGIC (-c config.yaml)
+
+Only MC:
+$ python merging_runs_and_splitting_training_samples.py --analysis-type onlyMC (-c config.yaml)
+"""
+
+import argparse
+import glob
+import logging
+import os
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+from tqdm import tqdm
+
+from magicctapipe import __version__
+
+__all__ = ["cleaning", "split_train_test", "merge", "mergeMC"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def cleaning(list_of_nodes, cwd):
+
+    """
+    This function looks for failed runs in each node and remove them.
+
+    Parameters
+    ----------
+    list_of_nodes : array of str
+        List of nodes where the function will look for failed runs.
+    cwd : Path
+        Current working directory
+    """
+
+    for i in tqdm(range(len(list_of_nodes)), desc="Cleaning failed runs"):
+        os.chdir(list_of_nodes[i])
+        os.system('find . -type f -name "*.h5" -size -1k -delete')
+
+    os.chdir(cwd)
+    print("Cleaning done.")
+
+
+def split_train_test(target_dir, train_fraction, source_name):
+
+    """
+    This function splits the MC proton sample in 2, i.e. the "test" and the "train" subsamples.
```

Comment:
this is not something that regular analyzer would do, normally you use all the protons for training.
This is only needed if you make performance MC-only studies. Maybe worth to add this in the description of the function?

Reply:
added